docs/articles/documentation/extending-testcafe/custom-reporter/creating-reporter-plugin/helper-methods.md

```diff
@@ -0,0 +1,297 @@
+---
+layout: docs
+title: Helper Methods
+permalink: /documentation/extending-testcafe/custom-reporter-plugin/helper-methods/
+---
+# Helper Methods
+
+The helper methods are used while [implementing a reporter](/testcafe/documentation/extending-testcafe/custom-reporter-plugin/creating-reporter-plugin/#implementing-reporter)
+to format a report output. TestCafe mixins these methods to the reporter.
+
+To refer to the methods, use `this`. The methods are chainable, which allows you to write compact code like this:
+
+```js
+this.setIndent(4)
+    .write(testname)
+    .newline()
+    .write(result)
+```
+
+## Formatting Helpers
+
+### indentString
+
+Indents each line in a string by a number of spaces.
+
+```text
+indentString (str, indentVal) → this
+```
+
+Parameter   | Type    | Description
+----------- | ------- | ------------------------------------------
+`str`       | String  | The string you want to indent.
+`indentVal` | Integer | The number of spaces to indent a new line.
+
+**Example**
+
+This example demonstrates how you can indent each line in the string `str` with four spaces.
+
+```js
+reportTaskStart (startTime, userAgents, testCount) {
+            ...
```

Comment:
Weird formatting. We use 4-spaces indentation. No need for this ellipsis here - the methods will be just fine without it.